### PR TITLE
[chains] Cycle httpx clients every 8h and upgrade poetry version.

### PR DESCRIPTION
**Pull request by marius-baseten**
## :microscope: Testing
Local integration tests.

### CODE REVIEW

File: truss-chains/truss_chains/stub.py [hunk 1]
```diff
@@ -12,6 +13,13 @@
 class BasetenSession:
     """Helper to invoke predict method on Baseten deployments."""
 
+    _client_cycle_time_sec: ClassVar[int] = 3600 * 8  # 8 hours.
```
Comment (squidarth) [inline truss-chains/truss_chains/stub.py:16]: should we just have this be a file-level constant?

**Review by squidarth — APPROVED** [review]
looks good. were you able to test this manually? if so could you write out in the PR description how? 

One more thing -- in a higher-concurrency environment, the second that the client "expires", every request will concurrently create a new httpx.Client. This is not particularly expensive, (and should be fine) right?

File: truss-chains/truss_chains/stub.py (hunk 1 above)
Comment (marius-baseten) [reply truss-chains/truss_chains/stub.py:16]: I'd make it all caps, but I think it makes sense to scope it to the `BasetenSession` session class only, there's no need to expose this outside.

**Comment by marius-baseten** [comment]
> looks good. were you able to test this manually? if so could you write out in the PR description how?
> 
> One more thing -- in a higher-concurrency environment, the second that the client "expires", every request will concurrently create a new httpx.Client. This is not particularly expensive, (and should be fine) right?

Good observation, I think I'll make an async lock around the re-creation.

**Pull request merged**